### ISSUE TITLE
Authentication failure

### ISSUE THREAD

**Issue by silvershadowcc** [matomo-org/matomo-php-tracker#70]
Just tried using the php tracker, but ran into an authentication failure when setting the visitor IP. The visit (with ua etc.) and title were logged, but with the web server IP.
As you can see I dumped $this after setting the auth token which showed the token and ip were properly set. They also appeared in the tracking url of the failure. I tried two auth tokens, one with super user access and one with site specific write access.

**Tracking failures - Administration**
```
Measurable
example.com (Id 4)

Problem
Request was not authenticated but authentication was required.

Solution
Set or correct a "token_auth" in your tracking request. Learn more
```

**PHP code** (from documentation example)
```
// Required variables
$matomoSiteId = 4;                  // Site ID
$matomoUrl = "https://example.com"; // Your matomo URL
$matomoToken = "";                  // Your authentication token

// Optional variable
$matomoPageTitle = self::$title;              // The title of the page

// Load object
require_once(__DIR__ . '/MatomoTracker.class.php');

// Matomo object
$mT = new MatomoTracker((int)$matomoSiteId, $matomoUrl);

// Set visitor IP
$mT->setIp(self::getIp());

// Set authentication token
print_r($mT->setTokenAuth($matomoToken));

// Track page view
$mT->doTrackPageView($matomoPageTitle);
```

**Comment by tsteur** [comment]
@silvershadowcc it's possible you got the failure initially but it's resolved now. That's because the tracking failure doesn't go away automatically once you send the correct request (it'll go away after a week or so). So maybe initially you were missing the token auth and then added it and it's working now? Does it get tracked correctly now or is it still using the webserver's IP? 

I assume `self::getIp()` returns the correct IP of the visitor?

**Comment by silvershadowcc** [comment]
@tsteur That might be the case. Is that due to the tracker or Matomo? And in the latter case; do you know what causes this, where the old request/token is cached(?) or whether there is some way to force this? I already retried while having the tracking failures removed. Atm I'm still getting the web server's IP and yes `self::getIp()` does return the correct IP hehe. I guess I still have to wait approximately five days then.

**Comment by tsteur** [comment]
If you already retried after tracking failures removed and the error appears again then something is still wrong. Maybe double check the configured token is 100% correct. Any chance you can look at the access logs on your server if the token auth is present in the URL? Does it work when you exec the same request in the browser maybe?

**Comment by silvershadowcc** [comment]
The configured token was correct. I went through my access log and found the URL that was used which contained that very token. Then I used it in my browser, it worked. The IP showed up as set in the URL. So, I executed the PHP script once more and now (without changing a thing) it did work as well. I suppose it really was just a matter of waiting for Matomo to clear the previous token value which might have been incorrect.

Thanks for the assistance.
